commit 9ecb333d56db4baf68b07887f9fbd5eae0698d89 (refs/changes/98/2998/3)
Author: Jonathan Perkin <jperkin@joyent.com>
Date:   2019-02-08T17:14:58+00:00 (8 months ago)
    
    OS-6469 CTF incorrect data object handling
    Reviewed by: Jason King <jason.king@joyent.com>
    Reviewed by: Robert Mustacchi <rm@joyent.com>
    Approved by: John Levon <john.levon@joyent.com>

diff --git a/usr/src/common/ctf/ctf_types.c b/usr/src/common/ctf/ctf_types.c
index 18d2d3d88a..95bb3f34f2 100644
--- a/usr/src/common/ctf/ctf_types.c
+++ b/usr/src/common/ctf/ctf_types.c
@@ -1083,6 +1083,8 @@ ctf_object_iter(ctf_file_t *fp, ctf_object_f *func, void *arg)
 			continue;
 		id = *(ushort_t *)((uintptr_t)fp->ctf_buf +
 		    fp->ctf_sxlate[i]);
+		if (id == 0)
+			continue;
 
 		/*
 		 * Validate whether or not we're looking at a data object as
